pkg/generate/app/strategyref.go

```diff
@@ -71,8 +71,17 @@ func (g *BuildStrategyRefGenerator) getSource(srcRef *SourceRef) error {
 	if srcRef.Dir, err = ioutil.TempDir("", "gen"); err != nil {
 		return err
 	}
-	if err = g.gitRepository.Clone(srcRef.Dir, srcRef.URL.String()); err != nil {
-		return fmt.Errorf("unable to clone repository at %s", srcRef.URL.String())
+	if len(srcRef.Ref) == 0 {
+		if err = g.gitRepository.CloneWithOptions(srcRef.Dir, srcRef.URL.String(), git.CloneOptions{
+			Recursive: true,
+			Shallow:   true,
+		}); err != nil {
+			return fmt.Errorf("unable to clone repository at %s", srcRef.URL.String())
+		}
+	} else {
+		if err = g.gitRepository.Clone(srcRef.Dir, srcRef.URL.String()); err != nil {
+			return fmt.Errorf("unable to clone repository at %s", srcRef.URL.String())
+		}
```

Comment:
imo we shouldn't have this exact block of code in two places (here and sourcelookup.go), seems like it could/should be a utility function, in case we continue to evolve this behavior or add other options.